dask/layers.py

```diff
@@ -37,9 +37,19 @@ def __call__(self, *args, **kwargs):
         return import_term(self.function_path)(*args, **kwargs)
 
 
-def nested_function(outer_func, inner_func, inputs):
-    """Nested-Function Utility"""
-    return outer_func(inner_func(inputs))
+def _compose(arg, *funcs):
+    """Nested-Function Utility
+
+    The first positional argument (`arg`) corresponds to
+    a single input argument to the inner-most function.
+    Other posional arguments (`*funcs`), correspond to
+    the nested functions in inner-to-outer order.
```

Comment:
To me this order is a little unintuitive. If you want `f(g(x))`, writing `compose(f, g, x)` is more intuitive than `compose(x, g, f)`. I think in functional languages, the composition operation is typically in `f, g, x` order:
* https://docs.julialang.org/en/v1/manual/functions/#Function-composition-and-piping
* https://wiki.haskell.org/Function_composition

Reply:
Although if you changed the name to `pipe`, then I think the order you have here would make sense. Maybe that's more what this is meant for? `pipe` I think makes more sense in general, because `compose` is usually just about function composition (returns a function); pipe returns a value (like this does).

Reply:
I do agree with the original order you suggested, but I had trouble getting `f(*funcs, x)`, while `f(x, *funcs)` worked right away.  From there, I only flipped the order of funcs, because it seemed wierd not to have the inner-most function be adjacent to the inputs :)

Now that you have pointed it out, this is indeed a simple `pipe` function, and it seems [functoolz already has what we need](https://toolz.readthedocs.io/en/latest/api.html#toolz.functoolz.pipe)